bugsnag-android-core/src/main/java/com/bugsnag/android/RemoteConfig.kt

```diff
@@ -0,0 +1,136 @@
+package com.bugsnag.android
+
+import android.util.JsonReader
+import com.bugsnag.android.RemoteConfig.Companion.DISCARD_RULE_ALL
+import com.bugsnag.android.RemoteConfig.Companion.DISCARD_RULE_ALL_HANDLED
+import com.bugsnag.android.RemoteConfig.Companion.MATCH_TYPE
+import com.bugsnag.android.internal.DateUtils
+import java.util.Date
+
+internal class RemoteConfig(
+    val configurationTag: String,
+    val configurationExpiry: Date,
+    val discardRules: List<DiscardRule>
+) : JsonStream.Streamable {
+
+    override fun toStream(stream: JsonStream) {
+        stream.beginObject()
+        stream.name(KEY_CONFIGURATION_TAG).value(configurationTag)
+        stream.name(KEY_CONFIG_EXPIRY).value(DateUtils.toIso8601(configurationExpiry))
+        stream.name(KEY_DISCARD_RULES).beginArray()
+        for (rule in discardRules) {
+            stream.beginObject()
+            when (rule) {
+                is DiscardRule.All -> stream.value(rule)
+                is DiscardRule.AllHandled -> stream.value(rule)
+            }
+            stream.endObject()
+        }
+        stream.endArray()
```

Comment:
`JsonStream` directly supports encoding `Collection`s as arrays
```suggestion
        stream.name(KEY_DISCARD_RULES).value(discardRules)
```